src/conn/conn_compact.c

```diff
@@ -549,8 +553,13 @@ __background_compact_server(void *arg)
             __wt_spin_unlock(session, &conn->background_compact.lock);
         }
 
-        /* When the entire metadata file has been parsed, take a break or wait until signalled. */
-        if (full_iteration || !running) {
+        /*
+         * Take a break or wait until signalled in the following conditions:
```

Comment:
Change "in the following conditions" to "in any of the following conditions"

Reply:
Thank you, see 8b097254fff78dae84013b9d4846d84f31e4423b